sdk/helper/ldaputil/client.go

```diff
@@ -75,10 +75,8 @@ func (c *Client) DialLDAP(cfg *ConfigEntry) (Connection, error) {
 			continue
 		}
 		if err == nil {
-			if retErr != nil {
-				if c.Logger.IsDebug() {
-					c.Logger.Debug("errors connecting to some hosts", "error", retErr.Error())
-				}
+			if retErr != nil && c.Logger != nil {
```

Comment:
should we have this logger nil check around all the calls to `c.Logger` in all the methods of Client? Or is it just required here?